OpenMage/magento-lts
# [Theme BC] Add form key validation to Contacts form

Add form key validation when submitting Contacts form.

### Description (*)

Similar to #1866, this PR adds form key validation to the Contacts form to prevent bots from sending spam requests to your email address. This option is disabled by default as it is a breaking change for custom themes.

With this change, the `hideit` field in contacts form is now obsolete, so it may be removed in a future PR if it's safe to do so.

### Fixed Issues (if relevant)

1. Fixes OpenMage/magento-lts#1911

### Manual testing scenarios (*)

1. Go to System -> Configuration -> General -> Contacts -> Security and enable form key validation.
2. Try submitting the form by issuing an HTTP request directly, if everything is fine the request should not go through. 
```sh
$ curl '<magento_host>/contacts/index/post' -d 'name=John+Doe&email=bot@example.org&comment=test'
```

### Contribution checklist (*)
